landlab/components/lateral_erosion/lateral_erosion.py

```diff
@@ -0,0 +1,631 @@
+            raise ValueError(
+                "inlet_on is True, but no inlet_node or inlet_area is provided."
+            )
+
+        if Kv is None:
+            raise ValueError(
+                "Kv must be set as a float, node array, or field name. It was None."
+            )
+
+        self._grid = grid
+        # Create fields needed for this component if not already existing
+        if "volume__lateral_erosion" in grid.at_node:
+            self.vol_lat = grid.at_node["volume__lateral_erosion"]
+        else:
+            self.vol_lat = grid.add_zeros("volume__lateral_erosion", at="node")
+        if "qs_in" in grid.at_node:
+            self.qs_in = grid.at_node["qs_in"]
+        else:
+            self.qs_in = grid.add_zeros("qs_in", at="node")
+
+        # you can specify the type of lateral erosion model you want to use.
+        # But if you don't the default is the undercutting-slump model
+        if latero_mech == "TB":
+            self._TB = True
+            self._UC = False
+        else:
+            self._UC = True
+            self._TB = False
+        # option use adaptive time stepping. Default is fixed dt supplied by user
+        if solver == "basic":
+            self.run_one_step = self.run_one_step_basic
+        elif solver == "adaptive":
+            self.run_one_step = self.run_one_step_adaptive
+            self.frac = 0.3  # for time step calculations
+        self.alph = alph
+        self.Kv = Kv  # can be overwritten with spatially variable
+        self.inlet_on = False  # will be overwritten below if inlet area is provided
+        self.Klr = float(Kl_ratio)  # default ratio of Kv/Kl is 1. Can be overwritten
+        self.wid_coeff = 0.4    #coefficient for calculating channel width
+        self.wid_exp = 0.35    #exponent for calculating channel width
+
+        self.dzdt = grid.add_zeros("dzdt", at="node", noclobber=False)  # elevation change rate (M/Y)
+        # optional inputs
+        self.inlet_on = inlet_on
+        if inlet_on:
+            self.inlet_node = inlet_node
+            self.inlet_area = inlet_area
+            # runoff is an array with values of the area of each node (dx**2)
+            runoffinlet = np.full(grid.number_of_nodes, grid.dx ** 2, dtype=float)
+            # Change the runoff at the inlet node to node area + inlet node
+            runoffinlet[inlet_node] += inlet_area
+            grid.add_field(
+                "water__unit_flux_in", runoffinlet, at="node", noclobber=False
+            )
+            # set qsinlet at inlet node. This doesn't have to be provided, defaults
+            # to 0.
+            self.qsinlet = qsinlet
+            self.qs_in[self.inlet_node] = self.qsinlet
+
+        # handling Kv for floats (inwhich case it populates an array N_nodes long) or
+        # for arrays of Kv. Checks that length of Kv array is good.
+        self.Kv = np.ones(self.grid.number_of_nodes, dtype=float) * Kv
+
+    def run_one_step_basic(self, dt=1.0):
```

Comment:
This function, and the adaptive counterpart could use a docstring. Just a quick statement saying what the function does and what parameters it takes.

Reply:
doc string added to the basic and adaptive time stepping functions.